adaptive_memory_v3.1.py

```
@@ -1659,79 +1711,85 @@ async def outlet(
+
+        # --- MESSAGE DE-DUPLICATION - ACQUIRE LOCK IMMEDIATELY ---
+        # Prevent multiple worker processes from processing the same message
+        # This MUST happen before any heavy processing (valve loading, background tasks, etc.)
+        message_hash = None
+        if last_user_message_content:
+            message_hash = self._create_message_hash(user_id, last_user_message_content)
+
+            if not await self._try_acquire_message_lock(message_hash):
+                logger.info(
+                    f"Outlet: Message {message_hash[:8]}... already being processed by another instance, "
+                    f"skipping duplicate outlet execution"
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)